File: Stack implementation through Array in C, JS and Ruby/stack_through_arrays.js
// Stack implementation using arrays in JavaScript

class Stack {
    constructor(size) {
      this.stack = new Array(size);
      this.top = -1;
      this.size = size;
    }
  
    // Push function to insert elements in the stack
    push(value) {
      if (this.top === this.size - 1) {
        console.log("Stack Overflow");
      } else {
        this.stack[++this.top] = value;
      }
    }
  
    // Pop function to remove elements from the stack
    pop() {
      if (this.top === -1) {
        console.log("Stack Underflow");
      } else {
        this.top--;
      }
    }
  
    // Show function to display elements in the stack
    show() {
      if (this.top === -1) {
        console.log("Stack is empty");
      } else {
        for (let i = this.top; i >= 0; i--) {
          console.log(this.stack[i]);
        }
      }
    }
  
    // Peek function to display the top element of the stack
    peek() {
      if (this.top === -1) {
        console.log("Stack Underflow");
        return null;
      } else {
        return this.stack[this.top];
      }
    }
  }
  
  // Main function to demonstrate stack operations
  function main() {
    const prompt = require('prompt-sync')({sigint: true});
    const n = parseInt(prompt("Enter the number of elements in the stack: "), 10);
    const stack = new Stack(n);
  
    console.log("*********Stack operations using array*********");
    console.log("----------------------------------------------");
  
    let choice = 0;
  
    while (choice !== 5) {
      console.log("\nChoose one from the below options...");
      console.log("1.Push\n2.Pop\n3.Show\n4.Peek\n5.Exit");
      choice = parseInt(prompt("Enter your choice: "), 10);
  
      switch (choice) {
        case 1:
          const value = parseInt(prompt("Enter the value: "), 10);
          stack.push(value);
          break;
        case 2:
          stack.pop();
          break;
        case 3:
          stack.show();
          break;
        case 4:
          console.log(stack.peek());
          break;
        case 5:
          console.log("Exiting....");
          break;
        default:
          console.log("Please Enter valid choice");
      }
    }
  }
  
  main();

File: Stack implementation through Array in C, JS and Ruby/stack_through_arrays.rb
# Implementtion of Stack through Arrays in Ruby

class Stack 
   
    def initialize(size)
        @stack = Array.new(size)
        @top = -1
        @size = size
    end

    # Push function to insert elements in the stack
    def push(value)
        if @top == @size - 1
            puts "Stack Overflow"
        else
            @top += 1
            @stack[@top] = value
        end
    end

    # Pop function to remove elements from the stack
    def pop 
        if @top == - 1
            puts "Stack Underflow"
        else 
            @top -= 1
        end
    end

    # Show function to display elements in the stack
    def show
        if @top == -1
            puts "Stack is Empty"
        else
            @top.downto(0) { |i| puts @stack[i] }
        end
    end

    # Peek function to display the top element of the stack
    def peek
    if @top == -1
        puts "Stack is Empty"
        nil
    else
        puts @stack[@top]
    end
    end

end

# Main function to demonstrate stack operations
def main
print "Enter the number of elements in the stack: "
size = gets.chomp.to_i
stack = Stack.new(size)

puts "*********Stack operations using array*********"
puts "----------------------------------------------"

choice = 0

# This program provides a menu-driven interface to perform stack operations.
while choice != 5
    puts "\nChoose one from the below options..."
    puts "1. Push"
    puts "2. Pop"
    puts "3. Show"
    puts "4. Peek"
    puts "5. Exit"
    print "Enter your choice: "
    choice = gets.chomp.to_i

    case choice
    when 1
        print "Enter the value: "
        value = gets.chomp.to_i
        stack.push(value)
    when 2
        stack.pop
    when 3
        stack.show
    when 4
        puts stack.peek
    when 5
        puts "Exiting..."
    else 
        puts "Invalid choice! Please Enter a valid choice..."
    end
end

end

main()